ios/RCTContacts/RCTContacts.m

```diff
@@ -61,41 +61,132 @@ - (NSDictionary *)constantsToExport
     }];
 }
 
-RCT_EXPORT_METHOD(getContactsMatchingString:(NSString *)string callback:(RCTResponseSenderBlock) callback)
+RCT_EXPORT_METHOD(getContactsMatchingString:(NSString *)string
+                             withThumbnails:(BOOL) withThumbnails
+                                   callback:(RCTResponseSenderBlock) callback
+                                   )
 {
     CNContactStore *contactStore = [[CNContactStore alloc] init];
     if (!contactStore)
         return;
-    [self getContactsFromAddressBook:contactStore matchingString:string callback:callback];
+    [self getContactsFromAddressBook:contactStore matchingString:string withThumbnails:withThumbnails callback:callback];
 }
 
 -(void) getContactsFromAddressBook:(CNContactStore *)store
                     matchingString:(NSString *)searchString
+                    withThumbnails:(BOOL) withThumbnails
                           callback:(RCTResponseSenderBlock)callback
 {
-    NSMutableArray *contacts = [[NSMutableArray alloc] init];
+    NSMutableArray<CNContact*> *contacts = [[NSMutableArray alloc] init];
     NSError *contactError = nil;
-    NSArray *keys = @[
-                      CNContactEmailAddressesKey,
-                      CNContactPhoneNumbersKey,
-                      CNContactFamilyNameKey,
-                      CNContactGivenNameKey,
-                      CNContactMiddleNameKey,
-                      CNContactPostalAddressesKey,
-                      CNContactOrganizationNameKey,
-                      CNContactJobTitleKey,
-                      CNContactImageDataAvailableKey,
-                      CNContactNoteKey,
-                      CNContactUrlAddressesKey,
-                      CNContactBirthdayKey
-                      ];
-    NSArray *arrayOfContacts = [store unifiedContactsMatchingPredicate:[CNContact predicateForContactsMatchingName:searchString]
-                                                           keysToFetch:keys
-                                                                 error:&contactError];
-    [arrayOfContacts enumerateObjectsUsingBlock:^(CNContact * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
-        NSDictionary *contactDictionary = [self contactToDictionary:obj withThumbnails:NO];
-        [contacts addObject:contactDictionary];
+
+    NSArray *nameKeys = @[
```

Comment:
These arrays can be pre-created since they are no longer a simple literal.

Reply:
When you say pre-created, do you mean construct them in the init so they don't need to get recreated every time the method gets called?

I'll definitely change the mutable array into a vanilla `NSArray` though